R/call_map_api

```diff
@@ -0,0 +1,454 @@
+#' # First find the key for birds on GBIF backbone taxonomy
+#' BirdBackboneKey <- name_backbone('aves')$classKey
+#' # Then run the function with standard settings
+#' BirdDiversityMap <- map_fetch(key = BirdBackboneKey)
+#' plot(BirdDiversityMap)
+#'
+#'
+#' # If we are interested in an overview of all animal specimen collected between
+#' # 1990 to 1930, we an use the query below to get a quick overview
+#' DivMap1 <- call_map_api(
+#'   key = 1,
+#'   decades = c('1900_1910','1910_1920','1920_1930'),
+#'   layers = 'SP',
+#'   living = FALSE,
+#'   fossil = FALSE,
+#'   breaks = c(1,3,5,10,30,50,100,300,500,1000)
+#' )
+#' plot(DivMap1)
+#'
+#' # Moreover, I can use the resulting raster to run some interesting analyses,
+#' # like, how does this compare to the equivalent records from the 1950 to 1980 period?
+#' DivMap2 <- call_map_api(
+#'   key = 1,
+#'   #decades = c('1950_1960','1960_1970','1970_1980'),
+#'   decades = c('1930_1940','1940_1950','1950_1960'),
+#'   layers = 'SP',
+#'   living = FALSE,
+#'   fossil = FALSE,
+#'   breaks = c(1,3,5,10,30,50,100,300,500,1000)
+#' )
+#' plot(DivMap2)
+#'
+#' # Since these are rasters, we can do calculations on them!
+#' plot(DivMap2 - DivMap1)
+#'
+#' # We can immediately see that more records were collected in North America,
+#' # but collection numbers went down in some areas of Europe, South-East Asia,
+#' # Australia, and Africa.
+#'
+#' # This is just one quick ad-hoc query.
+#' # You can run your own spatial analyses in a similar fashion to investigate
+#' # any question you might be interested in.}
+map_fetch <- function(
+  type = 'TAXON',
+  key = 1,
+  resolution = 1,
+  layers = c('OBS','SP','OTH'),
+  living = TRUE,
+  fossil = FALSE,
+  decades = c('NO_YEAR','PRE_1900','1900_1910','1910_1920','1920_1930',
+              '1930_1940','1940_1950','1950_1960','1960_1970','1970_1980',
+              '1980_1990','1990_2000','2000_2010','2010_2020'),
+  crs_string = '+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0',
+  map_api_path = '/v1/map/density/tile',
+  breaks = NULL,
+  nbreaks = NULL,
+  x = 0,
+  y = 0,
+  z = 0
+  ) {
+
+  # Check if required raster package is loaded
+  raster = TRUE
+  if(!'package:raster' %in% search()){
+    warning('"raster" package is recommended for this function, but was not found. The function will return the API call, but you will not be able to use the main data output properly. We recommend you install the raster package via install.packages("raster")!',
+      call. = FALSE)
+    raster = FALSE
+  }
+  # Check type of search. If other than 'TAXON', breaks cannot be set automatically
+  if(type != 'TAXON' & is.null(breaks)){
+    warning(paste(
+      'Automatic setting of breaks is not supported for type',
+      type,'. A standard scale will be used, but results will likely be better when supplying manual break values',
+      call. = FALSE)
+    )
+  }
+
+  # Check input arguments
+  if(!is.numeric(x) & !is.integer(x)){
```

Comment:
here and throughout, i think you want the double ampersand `&&`, which leads to a single logical. which you'd want if you're doing an `if` statement